index.js

```diff
@@ -0,0 +1,107 @@
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session);
+
+
+const knexConfig = require('./knexfile.js');
+const db = knex(knexConfig.development);
+
+const server = express();
+
+const sessionConfig = {
+    secret: 'Extremelysecretthinghere',
+    name: 'brad',
+    httpOnly: true,
+    saveUninitialized: false,
+    cookie: {
+        secure: false,
+        maxAge: 1000 * 60 * 10
+    },
+    store: new KnexSessionStore({
+        tablename: 'sessions',
+        sidfieldname: 'sid',
+        knex: db,
+        createtable: true,
+        clearInterval: 1000 * 60 * 60
+    })
+};
+server.use(session(sessionConfig));
+
+server.use(express.json());
+server.use(helmet());
+
+// tests to make sure server is working
+server.get('/', (req, res) => {
+  res.send('***Is your server running? you better catch it before it runs away ***')
+});
+// registers a new user
+server.post('/api/register', (req, res) => {
+    const credentials = req.body;
+    const hash = bcrypt.hashSync(credentials.password, 14);
+    credentials.password = hash;
+    db('users')
+    .insert(credentials)
+    .then(ids => {
+        const id = ids[0];
+        req.session.userId = id;
+        res.status(201).json({ newUserId: id});
+    })
+    .catch(err => {
+        res.status(500).json(err);
+    });
+});
+
+server.post('/api/login', (req,res) => {
+    const creds = req.body;
+    db('users').where({username: creds.username}).first()
+    .then(user => {
+        if(user && bcrypt.compareSync(creds.password, user.password)) {
```

Comment:
Sometimes it is a good idea to delegate bcrypt methods to a separate middleware file, create a function and call it in index after importing.